django/utils/timezone.py

```diff
@@ -41,7 +41,21 @@
 NOT_PASSED = object()
 
 
-utc = timezone.utc
+def __getattr__(name):
+    if name != "utc":
+        raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
+
+    warnings.warn(
+        "The django.utils.timezone.utc alias is deprecated. "
+        "Please update your code to use datetime.timezone.utc instead.",
+        RemovedInDjango50Warning,
+    )
+
+    return timezone.utc
+
+
+def __dir__():
+    return sorted(list(globals()) + ["utc"])
```

Comment:
I'm not sure about using `globals()` :thinking: What do you think about moving `__dir__()` at the end of the file and using `dir()` instead of `globals()`:
```python
# RemovedInDjango50Warning.
_DIR = dir()

def __dir__():
    return sorted([*_DIR, "utc"])
```
we would avoid adding `_DIR`, `__warningregistry__`, and `__dir__` to the `dir(timezone)`.

Reply:
Meh… I just went with [the Python Core Contributor Recommended Implementation™](https://github.com/django/django/pull/14848#discussion_r708378262) 😃

Reply:
It's just a suggestion. Deprecation is tricky for module constants, IMO we should try to avoid side-effects.